src/main.rs

```diff
@@ -186,6 +177,7 @@ async fn handle_request(
     rfb_addr: std::net::SocketAddr,
     mut req: Request<Body>,
     remote_addr: SocketAddr,
+    authentication: Arc<auth::RfbAuthentication>,
```

Comment:
Could we make `RfbAuthentication` derive `Clone` and `Copy` so we don't have to use an `Arc` here?
This should also remove the need for any `authentication.clone()`s

Reply:
i'm going to add a struct that has a https://doc.rust-lang.org/std/collections/struct.HashMap.html in another change. that one has `Clone`, but isn't `Copy` so IIUC i cannot legally make it `Copy` too :(

Reply:
Oh yup, that'll need the `Arc`

Reply:
yup, just tried that and the compiler balked at me:

```
error[E0204]: the trait `Copy` may not be implemented for this type
  --> src/auth.rs:13:10
   |
13 | #[derive(Copy, Clone)]
   |          ^^^^
...
34 |         replid: String,
   |         -------------- this field does not implement `Copy`
35 |         pubkeys: HashMap<String, Vec<u8>>,
   |         --------------------------------- this field does not implement `Copy`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```